modin/backends/pandas/query_compiler.py

```diff
@@ -1574,6 +1574,20 @@ def compute_groupby(df):
             axis, lambda df: groupby_agg_builder(df)
         )
         result = self.__constructor__(new_modin_frame)
+
+        # that means that exception in `compute_groupby` was raised
+        # in every partition, so we also should raise it
+        if len(result.columns) == 0 and len(self.columns) != 0:
```

Comment:
If this is an issue that only happens for numeric operations, can we make this check before we do any compute?

Reply:
I'm afraid not, different aggregation functions raise exceptions at different non-numeric types. Look at this example:
```python
import pandas
import datetime

data = {"by": [1, 2, 3, 2, 1], "date": [datetime.datetime(2020, 2, 6)] * 5}

grp = pandas.DataFrame(data).groupby("by")

print(grp.quantile())  # prints the result of quantile
print(grp.mean())  # DataError: No numeric types to aggregate
```

Reply:
I see. For most cases, I do not expect this to impact performance.